lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -956,55 +956,75 @@ namespace Js
             JavascriptError::ThrowTypeError(GetScriptContext(), JSERR_ErrorOnRevokedProxy, _u("ownKeys"));
         }
 
-        Var propertyName = nullptr;
-        PropertyId propertyId;
-        int index = 0;
-        JsUtil::BaseDictionary<const char16*, Var, Recycler> dict(requestContext->GetRecycler());
-        JavascriptArray* arrResult = requestContext->GetLibrary()->CreateArray();
-
-        // 13.7.5.15 EnumerateObjectProperties(O) (https://tc39.github.io/ecma262/#sec-enumerate-object-properties)
-        // for (let key of Reflect.ownKeys(obj)) {
-        Var trapResult = JavascriptOperators::GetOwnPropertyNames(this, requestContext);
-        if (JavascriptArray::Is(trapResult))
+        struct ProxyOwnkeysEnumerator : public JavascriptEnumerator
         {
-            JavascriptStaticEnumerator trapEnumerator;
-            if (!((JavascriptArray*)trapResult)->GetEnumerator(&trapEnumerator, EnumeratorFlags::SnapShotSemantics, requestContext))
+            typedef JsUtil::BaseHashSet<const char16*, Recycler> VisitedNamesHashSet;
+            Field(VisitedNamesHashSet*) visited;
+            Field(JavascriptArray*) trapResult;
+            Field(JavascriptProxy*) proxy;
+            FieldNoBarrier(ScriptContext*) scriptContext;
+            Field(JavascriptStaticEnumerator) innerEnumerator;
+            Field(uint32) index;
+
+            DEFINE_VTABLE_CTOR_ABSTRACT(ProxyOwnkeysEnumerator, JavascriptEnumerator)
+
+            ProxyOwnkeysEnumerator(ScriptContext* scriptContext, JavascriptProxy* proxy, JavascriptArray* trapResult)
+                :JavascriptEnumerator(scriptContext), scriptContext(scriptContext), proxy(proxy), trapResult(trapResult)
             {
-                return FALSE;
+                visited = RecyclerNew(scriptContext->GetRecycler(), VisitedNamesHashSet, scriptContext->GetRecycler());
             }
-            while ((propertyName = trapEnumerator.MoveAndGetNext(propertyId)) != NULL)
+            JavascriptStaticEnumerator* GetInnerEnumerator() { return &innerEnumerator; }
+            virtual uint32 GetCurrentItemIndex()  override { return index; }
+            virtual void Reset() override
```

Comment:
You don't have to override this.  This is only neededfor JSONParser to quickly determine if the value of an array index.    You are not returning an array index here.